File: backend/app/routes/graduate.py
"""
考研院校信息API路由
"""
from flask import Blueprint, request, jsonify
from app.models.graduate_school import (
    GraduateSchool, GraduateMajor, ScoreLine, ExamSubject
)
from app import db

graduate_bp = Blueprint('graduate', __name__)


@graduate_bp.route('/schools', methods=['GET'])
def get_schools():
    """获取院校列表"""
    page = request.args.get('page', 1, type=int)
    page_size = request.args.get('page_size', 20, type=int)
    province = request.args.get('province')
    level = request.args.get('level')  # 985/211/双一流
    school_type = request.args.get('type')
    keyword = request.args.get('keyword')
    
    query = GraduateSchool.query
    
    if province:
        query = query.filter(GraduateSchool.province == province)
    if level:
        if level == '985':
            query = query.filter(GraduateSchool.is_985 == True)
        elif level == '211':
            query = query.filter(GraduateSchool.is_211 == True)
        elif level == '双一流':
            query = query.filter(GraduateSchool.is_double_first == True)
    if school_type:
        query = query.filter(GraduateSchool.type == school_type)
    if keyword:
        query = query.filter(GraduateSchool.name.like(f'%{keyword}%'))
    
    pagination = query.order_by(GraduateSchool.is_985.desc(), 
                                GraduateSchool.is_211.desc(),
                                GraduateSchool.name).paginate(
        page=page, per_page=page_size, error_out=False
    )
    
    return jsonify({
        'success': True,
        'data': {
            'records': [s.to_dict() for s in pagination.items],
            'total': pagination.total,
            'page': page,
            'page_size': page_size,
            'pages': pagination.pages
        }
    })


@graduate_bp.route('/schools/<int:school_id>', methods=['GET'])
def get_school_detail(school_id):
    """获取院校详情"""
    school = GraduateSchool.query.get_or_404(school_id)
    data = school.to_dict()
    data['major_count'] = school.majors.count()
    return jsonify({
        'success': True,
        'data': data
    })


@graduate_bp.route('/schools/<int:school_id>/majors', methods=['GET'])
def get_school_majors(school_id):
    """获取院校的专业列表"""
    page = request.args.get('page', 1, type=int)
    page_size = request.args.get('page_size', 20, type=int)
    category = request.args.get('category')
    degree_type = request.args.get('degree_type')
    keyword = request.args.get('keyword')
    
    query = GraduateMajor.query.filter(GraduateMajor.school_id == school_id)
    
    if category:
        query = query.filter(GraduateMajor.category == category)
    if degree_type:
        query = query.filter(GraduateMajor.degree_type == degree_type)
    if keyword:
        query = query.filter(GraduateMajor.name.like(f'%{keyword}%'))
    
    pagination = query.order_by(GraduateMajor.code).paginate(
        page=page, per_page=page_size, error_out=False
    )
    
    return jsonify({
        'success': True,
        'data': {
            'records': [m.to_dict() for m in pagination.items],
            'total': pagination.total,
            'page': page,
            'page_size': page_size
        }
    })


@graduate_bp.route('/majors/<int:major_id>', methods=['GET'])
def get_major_detail(major_id):
    """获取专业详情"""
    major = GraduateMajor.query.get_or_404(major_id)
    return jsonify({
        'success': True,
        'data': major.to_dict()
    })


@graduate_bp.route('/majors/<int:major_id>/score-lines', methods=['GET'])
def get_major_score_lines(major_id):
    """获取专业的历年分数线"""
    score_lines = ScoreLine.query.filter(
        ScoreLine.major_id == major_id
    ).order_by(ScoreLine.year.desc()).all()
    
    return jsonify({
        'success': True,
        'data': [s.to_dict() for s in score_lines]
    })


@graduate_bp.route('/majors/<int:major_id>/exam-subjects', methods=['GET'])
def get_major_exam_subjects(major_id):
    """获取专业的考试科目"""
    year = request.args.get('year', type=int)
    
    query = ExamSubject.query.filter(ExamSubject.major_id == major_id)
    if year:
        query = query.filter(ExamSubject.year == year)
    else:
        # 默认获取最新年份
        latest = ExamSubject.query.filter(
            ExamSubject.major_id == major_id
        ).order_by(ExamSubject.year.desc()).first()
        if latest:
            query = query.filter(ExamSubject.year == latest.year)
    
    subjects = query.order_by(ExamSubject.subject_type).all()
    
    return jsonify({
        'success': True,
        'data': [s.to_dict() for s in subjects]
    })


@graduate_bp.route('/score-lines/search', methods=['GET'])
def search_score_lines():
    """搜索分数线"""
    page = request.args.get('page', 1, type=int)
    page_size = request.args.get('page_size', 20, type=int)
    school_name = request.args.get('school_name')
    major_name = request.args.get('major_name')
    year = request.args.get('year', type=int)
    province = request.args.get('province')
    min_score = request.args.get('min_score', type=int)
    max_score = request.args.get('max_score', type=int)
    
    query = ScoreLine.query.join(GraduateMajor).join(GraduateSchool)
    
    if school_name:
        query = query.filter(GraduateSchool.name.like(f'%{school_name}%'))
    if major_name:
        query = query.filter(GraduateMajor.name.like(f'%{major_name}%'))
    if year:
        query = query.filter(ScoreLine.year == year)
    if province:
        query = query.filter(GraduateSchool.province == province)
    if min_score:
        query = query.filter(ScoreLine.total_score >= min_score)
    if max_score:
        query = query.filter(ScoreLine.total_score <= max_score)
    
    pagination = query.order_by(ScoreLine.year.desc(), 
                                ScoreLine.total_score.desc()).paginate(
        page=page, per_page=page_size, error_out=False
    )
    
    return jsonify({
        'success': True,
        'data': {
            'records': [s.to_dict() for s in pagination.items],
            'total': pagination.total,
            'page': page,
            'page_size': page_size
        }
    })


@graduate_bp.route('/provinces', methods=['GET'])
def get_provinces():
    """获取所有省份列表"""
    provinces = db.session.query(GraduateSchool.province).distinct().all()
    return jsonify({
        'success': True,
        'data': [p[0] for p in provinces if p[0]]
    })


@graduate_bp.route('/categories', methods=['GET'])
def get_categories():
    """获取所有学科门类"""
    categories = db.session.query(GraduateMajor.category).distinct().all()
    return jsonify({
        'success': True,
        'data': [c[0] for c in categories if c[0]]
    })


@graduate_bp.route('/estimate-score', methods=['POST'])
def estimate_score():
    """分数估算 - 根据用户成绩估算可报考的学校"""
    data = request.get_json()
    total_score = data.get('total_score')
    politics_score = data.get('politics_score')
    english_score = data.get('english_score')
    math_score = data.get('math_score')
    professional_score = data.get('professional_score')
    category = data.get('category')  # 学科门类
    province = data.get('province')  # 目标省份
    
    if not total_score:
        return jsonify({'success': False, 'message': '请输入总分'}), 400
    
    # 查询符合条件的分数线（最近3年）
    query = ScoreLine.query.join(GraduateMajor).join(GraduateSchool)
    query = query.filter(ScoreLine.year >= 2022)
    query = query.filter(ScoreLine.total_score <= total_score)
    
    if politics_score:
        query = query.filter(ScoreLine.politics_score <= politics_score)
    if english_score:
        query = query.filter(ScoreLine.english_score <= english_score)
    if category:
        query = query.filter(GraduateMajor.category == category)
    if province:
        query = query.filter(GraduateSchool.province == province)
    
    # 按分数线从高到低排序，取前50个
    results = query.order_by(ScoreLine.total_score.desc()).limit(50).all()
    
    # 分类：冲刺/稳妥/保底
    score_diff = 20  # 分数差阈值
    categorized = {
        'reach': [],    # 冲刺（分数线接近用户分数）
        'match': [],    # 稳妥（分数线低于用户分数10-30分）
        'safe': []      # 保底（分数线低于用户分数30分以上）
    }
    
    for sl in results:
        diff = total_score - sl.total_score
        item = sl.to_dict()
        if diff <= 10:
            categorized['reach'].append(item)
        elif diff <= 30:
            categorized['match'].append(item)
        else:
            categorized['safe'].append(item)
    
    return jsonify({
        'success': True,
        'data': {
            'user_score': total_score,
            'reach': categorized['reach'][:10],
            'match': categorized['match'][:15],
            'safe': categorized['safe'][:10]
        }
    })


# ============ 院校管理接口 ============

@graduate_bp.route('/schools', methods=['POST'])
def create_school():
    """添加院校"""
    data = request.get_json()
    
    # 验证必填字段
    if not data.get('name'):
        return jsonify({'success': False, 'message': '院校名称不能为空'}), 400
    if not data.get('code'):
        return jsonify({'success': False, 'message': '院校代码不能为空'}), 400
    
    # 检查是否已存在
    existing = GraduateSchool.query.filter_by(code=data['code']).first()
    if existing:
        return jsonify({'success': False, 'message': '该院校代码已存在'}), 400
    
    school = GraduateSchool(
        name=data['name'],
        code=data['code'],
        province=data.get('province'),
        city=data.get('city'),
        type=data.get('type'),
        level=data.get('level'),
        is_985=data.get('is_985', False),
        is_211=data.get('is_211', False),
        is_double_first=data.get('is_double_first', False),
        website=data.get('website'),
        phone=data.get('phone'),
        address=data.get('address'),
        description=data.get('description')
    )
    
    db.session.add(school)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '添加成功',
        'data': school.to_dict()
    })


@graduate_bp.route('/schools/<int:school_id>', methods=['PUT'])
def update_school(school_id):
    """更新院校信息"""
    school = GraduateSchool.query.get_or_404(school_id)
    data = request.get_json()
    
    # 更新字段
    if 'name' in data:
        school.name = data['name']
    if 'code' in data:
        school.code = data['code']
    if 'province' in data:
        school.province = data['province']
    if 'city' in data:
        school.city = data['city']
    if 'type' in data:
        school.type = data['type']
    if 'level' in data:
        school.level = data['level']
    if 'is_985' in data:
        school.is_985 = data['is_985']
    if 'is_211' in data:
        school.is_211 = data['is_211']
    if 'is_double_first' in data:
        school.is_double_first = data['is_double_first']
    if 'website' in data:
        school.website = data['website']
    if 'phone' in data:
        school.phone = data['phone']
    if 'address' in data:
        school.address = data['address']
    if 'description' in data:
        school.description = data['description']
    
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '更新成功',
        'data': school.to_dict()
    })


@graduate_bp.route('/schools/<int:school_id>', methods=['DELETE'])
def delete_school(school_id):
    """删除院校"""
    school = GraduateSchool.query.get_or_404(school_id)
    
    # 删除关联的专业、分数线、考试科目
    for major in school.majors:
        ScoreLine.query.filter_by(major_id=major.id).delete()
        ExamSubject.query.filter_by(major_id=major.id).delete()
    GraduateMajor.query.filter_by(school_id=school_id).delete()
    
    db.session.delete(school)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '删除成功'
    })


# ============ 专业管理接口 ============

@graduate_bp.route('/majors', methods=['POST'])
def create_major():
    """添加专业"""
    data = request.get_json()
    
    if not data.get('school_id'):
        return jsonify({'success': False, 'message': '请选择所属院校'}), 400
    if not data.get('name'):
        return jsonify({'success': False, 'message': '专业名称不能为空'}), 400
    if not data.get('code'):
        return jsonify({'success': False, 'message': '专业代码不能为空'}), 400
    
    major = GraduateMajor(
        school_id=data['school_id'],
        name=data['name'],
        code=data['code'],
        category=data.get('category'),
        degree_type=data.get('degree_type', '学术型'),
        duration=data.get('duration', 3),
        department=data.get('department'),
        research_directions=data.get('research_directions'),
        description=data.get('description')
    )
    
    db.session.add(major)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '添加成功',
        'data': major.to_dict()
    })


@graduate_bp.route('/majors/<int:major_id>', methods=['PUT'])
def update_major(major_id):
    """更新专业信息"""
    major = GraduateMajor.query.get_or_404(major_id)
    data = request.get_json()
    
    if 'name' in data:
        major.name = data['name']
    if 'code' in data:
        major.code = data['code']
    if 'category' in data:
        major.category = data['category']
    if 'degree_type' in data:
        major.degree_type = data['degree_type']
    if 'duration' in data:
        major.duration = data['duration']
    if 'department' in data:
        major.department = data['department']
    if 'research_directions' in data:
        major.research_directions = data['research_directions']
    if 'description' in data:
        major.description = data['description']
    
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '更新成功',
        'data': major.to_dict()
    })


@graduate_bp.route('/majors/<int:major_id>', methods=['DELETE'])
def delete_major(major_id):
    """删除专业"""
    major = GraduateMajor.query.get_or_404(major_id)
    
    # 删除关联的分数线和考试科目
    ScoreLine.query.filter_by(major_id=major_id).delete()
    ExamSubject.query.filter_by(major_id=major_id).delete()
    
    db.session.delete(major)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '删除成功'
    })


# ============ 分数线管理接口 ============

@graduate_bp.route('/score-lines', methods=['POST'])
def create_score_line():
    """添加分数线"""
    data = request.get_json()
    
    if not data.get('major_id'):
        return jsonify({'success': False, 'message': '请选择专业'}), 400
    if not data.get('year'):
        return jsonify({'success': False, 'message': '请选择年份'}), 400
    if not data.get('total_score'):
        return jsonify({'success': False, 'message': '请输入总分线'}), 400
    
    score_line = ScoreLine(
        major_id=data['major_id'],
        year=data['year'],
        total_score=data['total_score'],
        politics_score=data.get('politics_score'),
        english_score=data.get('english_score'),
        math_score=data.get('math_score'),
        professional_score=data.get('professional_score'),
        enrollment_num=data.get('enrollment_num'),
        applicant_num=data.get('applicant_num'),
        admission_ratio=data.get('admission_ratio'),
        remark=data.get('remark')
    )
    
    db.session.add(score_line)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '添加成功',
        'data': score_line.to_dict()
    })


@graduate_bp.route('/score-lines/<int:score_line_id>', methods=['PUT'])
def update_score_line(score_line_id):
    """更新分数线"""
    score_line = ScoreLine.query.get_or_404(score_line_id)
    data = request.get_json()
    
    if 'year' in data:
        score_line.year = data['year']
    if 'total_score' in data:
        score_line.total_score = data['total_score']
    if 'politics_score' in data:
        score_line.politics_score = data['politics_score']
    if 'english_score' in data:
        score_line.english_score = data['english_score']
    if 'math_score' in data:
        score_line.math_score = data['math_score']
    if 'professional_score' in data:
        score_line.professional_score = data['professional_score']
    if 'enrollment_num' in data:
        score_line.enrollment_num = data['enrollment_num']
    if 'applicant_num' in data:
        score_line.applicant_num = data['applicant_num']
    if 'admission_ratio' in data:
        score_line.admission_ratio = data['admission_ratio']
    if 'remark' in data:
        score_line.remark = data['remark']
    
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '更新成功',
        'data': score_line.to_dict()
    })


@graduate_bp.route('/score-lines/<int:score_line_id>', methods=['DELETE'])
def delete_score_line(score_line_id):
    """删除分数线"""
    score_line = ScoreLine.query.get_or_404(score_line_id)
    db.session.delete(score_line)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '删除成功'
    })


# ============ 考试科目管理接口 ============

@graduate_bp.route('/exam-subjects', methods=['POST'])
def create_exam_subject():
    """添加考试科目"""
    data = request.get_json()
    
    if not data.get('major_id'):
        return jsonify({'success': False, 'message': '请选择专业'}), 400
    if not data.get('subject_name'):
        return jsonify({'success': False, 'message': '请输入科目名称'}), 400
    
    subject = ExamSubject(
        major_id=data['major_id'],
        year=data.get('year', 2024),
        subject_code=data.get('subject_code'),
        subject_name=data['subject_name'],
        subject_type=data.get('subject_type', '专业课'),
        full_score=data.get('full_score', 150),
        reference_books=data.get('reference_books'),
        remark=data.get('remark')
    )
    
    db.session.add(subject)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '添加成功',
        'data': subject.to_dict()
    })


@graduate_bp.route('/exam-subjects/<int:subject_id>', methods=['PUT'])
def update_exam_subject(subject_id):
    """更新考试科目"""
    subject = ExamSubject.query.get_or_404(subject_id)
    data = request.get_json()
    
    if 'year' in data:
        subject.year = data['year']
    if 'subject_code' in data:
        subject.subject_code = data['subject_code']
    if 'subject_name' in data:
        subject.subject_name = data['subject_name']
    if 'subject_type' in data:
        subject.subject_type = data['subject_type']
    if 'full_score' in data:
        subject.full_score = data['full_score']
    if 'reference_books' in data:
        subject.reference_books = data['reference_books']
    if 'remark' in data:
        subject.remark = data['remark']
    
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '更新成功',
        'data': subject.to_dict()
    })


@graduate_bp.route('/exam-subjects/<int:subject_id>', methods=['DELETE'])
def delete_exam_subject(subject_id):
    """删除考试科目"""
    subject = ExamSubject.query.get_or_404(subject_id)
    db.session.delete(subject)
    db.session.commit()
    
    return jsonify({
        'success': True,
        'message': '删除成功'
    })

File: frontend/src/api/graduate.js
/**
 * 考研院校信息API
 */
import request from '../utils/request'

// ============ AI 智能推荐配置 ============
// TODO: 后续替换为真实的 AI 接口地址
const AI_GRADUATE_CONFIG = {
  // AI 院校推荐接口地址（预留）
  recommendEndpoint: '/ai/graduate/recommend',
  // AI 择校分析接口地址（预留）
  analysisEndpoint: '/ai/graduate/analysis',
  // AI 备考建议接口地址（预留）
  adviceEndpoint: '/ai/graduate/advice',
  // 是否使用模拟数据（开发阶段使用）
  useMock: true
}

/**
 * AI 智能院校推荐
 * @param {Object} params - 推荐参数
 * @param {number} params.total_score - 总分
 * @param {number} params.politics_score - 政治分数
 * @param {number} params.english_score - 英语分数
 * @param {number} params.math_score - 数学分数
 * @param {number} params.professional_score - 专业课分数
 * @param {string} params.category - 学科门类
 * @param {string} params.province - 目标省份
 * @param {string} params.preference - 偏好（如：985优先、地域优先等）
 * @returns {Promise}
 */
export function getAIRecommendation(params = {}) {
  if (AI_GRADUATE_CONFIG.useMock) {
    return mockAIRecommendation(params)
  }
  
  return request({
    url: AI_GRADUATE_CONFIG.recommendEndpoint,
    method: 'post',
    data: params
  })
}

/**
 * AI 择校分析
 * @param {Object} params - 分析参数
 * @param {number} params.school_id - 目标院校ID
 * @param {number} params.major_id - 目标专业ID
 * @param {Object} params.scores - 用户成绩
 * @returns {Promise}
 */
export function getAISchoolAnalysis(params = {}) {
  if (AI_GRADUATE_CONFIG.useMock) {
    return mockAISchoolAnalysis(params)
  }
  
  return request({
    url: AI_GRADUATE_CONFIG.analysisEndpoint,
    method: 'post',
    data: params
  })
}

/**
 * AI 备考建议
 * @param {Object} params - 参数
 * @param {Object} params.scores - 当前成绩
 * @param {Object} params.target - 目标院校/专业
 * @param {number} params.months - 备考月数
 * @returns {Promise}
 */
export function getAIStudyAdvice(params = {}) {
  if (AI_GRADUATE_CONFIG.useMock) {
    return mockAIStudyAdvice(params)
  }
  
  return request({
    url: AI_GRADUATE_CONFIG.adviceEndpoint,
    method: 'post',
    data: params
  })
}

/**
 * 模拟 AI 院校推荐
 */
function mockAIRecommendation(params) {
  return new Promise((resolve) => {
    setTimeout(() => {
      const score = params.total_score || 350
      resolve({
        success: true,
        data: {
          summary: `根据您的成绩（总分 ${score} 分），AI 为您分析了全国 ${score > 380 ? '985/211' : '双一流及普通'}院校的录取情况。`,
          recommendation: generateRecommendationText(params),
          confidence: score > 400 ? 85 : score > 350 ? 70 : 55,
          suggestions: [
            score > 380 ? '您的成绩较为优秀，可以冲刺顶尖院校' : '建议以稳妥院校为主，适当冲刺',
            '建议关注目标院校的调剂信息',
            '复试准备要提前开始，尤其是英语口语'
          ]
        }
      })
    }, 1500)
  })
}

/**
 * 生成推荐文本
 */
function generateRecommendationText(params) {
  const score = params.total_score || 350
  const category = params.category || '工学'
  const province = params.province || ''
  
  let text = `## 📊 AI 智能择校分析报告\n\n`
  text += `### 一、成绩评估\n`
  text += `您的总分 **${score}** 分，`
  
  if (score >= 400) {
    text += `属于高分段，在${category}类专业中具有较强竞争力。\n\n`
    text += `### 二、院校推荐策略\n`
    text += `- **冲刺目标**：清华、北大、复旦等顶尖985院校\n`
    text += `- **稳妥选择**：中上游985或顶尖211院校\n`
    text += `- **保底方案**：普通211或强势双一流院校\n\n`
  } else if (score >= 350) {
    text += `属于中等偏上水平，建议采取"稳中求进"策略。\n\n`
    text += `### 二、院校推荐策略\n`
    text += `- **冲刺目标**：中等985或顶尖211院校\n`
    text += `- **稳妥选择**：普通211或强势双一流院校\n`
    text += `- **保底方案**：普通双一流或省属重点院校\n\n`
  } else {
    text += `建议以稳妥为主，重点关注调剂机会。\n\n`
    text += `### 二、院校推荐策略\n`
    text += `- **冲刺目标**：普通211院校\n`
    text += `- **稳妥选择**：双一流或省属重点院校\n`
    text += `- **保底方案**：普通院校，关注调剂\n\n`
  }
  
  text += `### 三、备考建议\n`
  text += `1. **专业课**：${category}类专业课是拉分关键，建议重点突破\n`
  text += `2. **公共课**：政治英语要保证过线，争取高分\n`
  text += `3. **复试准备**：提前准备英语口语和专业面试\n`
  
  if (province) {
    text += `\n### 四、地域分析\n`
    text += `您选择的目标省份是 **${province}**，该地区院校竞争${['北京', '上海', '江苏', '浙江'].includes(province) ? '较为激烈' : '相对适中'}。\n`
  }
  
  text += `\n---\n*以上分析基于历年数据，仅供参考。实际录取情况请以各院校官方公布为准。*`
  
  return text
}

/**
 * 模拟 AI 择校分析
 */
function mockAISchoolAnalysis(params) {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve({
        success: true,
        data: {
          matchScore: Math.floor(Math.random() * 30) + 60,
          analysis: '根据您的成绩和目标院校历年录取数据分析，您有一定的录取机会。建议重点提升专业课成绩。',
          risks: ['竞争激烈', '复试比例较高'],
          advantages: ['成绩达到往年分数线', '专业课基础扎实']
        }
      })
    }, 1000)
  })
}

/**
 * 模拟 AI 备考建议
 */
function mockAIStudyAdvice(params) {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve({
        success: true,
        data: {
          plan: '建议制定详细的复习计划，每天保证8小时有效学习时间。',
          priorities: ['专业课强化', '英语阅读提升', '政治冲刺'],
          timeline: [
            { month: '1-3月', task: '基础阶段，夯实基础知识' },
            { month: '4-6月', task: '强化阶段，重点突破难点' },
            { month: '7-9月', task: '提高阶段，大量刷题' },
            { month: '10-12月', task: '冲刺阶段，模拟考试' }
          ]
        }
      })
    }, 1000)
  })
}

// 导出配置供外部使用
export { AI_GRADUATE_CONFIG }

// 获取院校列表
export function getSchools(params) {
  return request({
    url: '/graduate/schools',
    method: 'get',
    params
  })
}

// 获取院校详情
export function getSchoolDetail(schoolId) {
  return request({
    url: `/graduate/schools/${schoolId}`,
    method: 'get'
  })
}

// 获取院校的专业列表
export function getSchoolMajors(schoolId, params) {
  return request({
    url: `/graduate/schools/${schoolId}/majors`,
    method: 'get',
    params
  })
}

// 获取专业详情
export function getMajorDetail(majorId) {
  return request({
    url: `/graduate/majors/${majorId}`,
    method: 'get'
  })
}

// 获取专业的历年分数线
export function getMajorScoreLines(majorId) {
  return request({
    url: `/graduate/majors/${majorId}/score-lines`,
    method: 'get'
  })
}

// 获取专业的考试科目
export function getMajorExamSubjects(majorId, params) {
  return request({
    url: `/graduate/majors/${majorId}/exam-subjects`,
    method: 'get',
    params
  })
}

// 搜索分数线
export function searchScoreLines(params) {
  return request({
    url: '/graduate/score-lines/search',
    method: 'get',
    params
  })
}

// 获取省份列表
export function getProvinces() {
  return request({
    url: '/graduate/provinces',
    method: 'get'
  })
}

// 获取学科门类列表
export function getCategories() {
  return request({
    url: '/graduate/categories',
    method: 'get'
  })
}

// 分数估算
export function estimateScore(data) {
  return request({
    url: '/graduate/estimate-score',
    method: 'post',
    data
  })
}


// ============ 院校管理接口 ============

// 添加院校
export function createSchool(data) {
  return request({
    url: '/graduate/schools',
    method: 'post',
    data
  })
}

// 更新院校
export function updateSchool(schoolId, data) {
  return request({
    url: `/graduate/schools/${schoolId}`,
    method: 'put',
    data
  })
}

// 删除院校
export function deleteSchool(schoolId) {
  return request({
    url: `/graduate/schools/${schoolId}`,
    method: 'delete'
  })
}

// ============ 专业管理接口 ============

// 添加专业
export function createMajor(data) {
  return request({
    url: '/graduate/majors',
    method: 'post',
    data
  })
}

// 更新专业
export function updateMajor(majorId, data) {
  return request({
    url: `/graduate/majors/${majorId}`,
    method: 'put',
    data
  })
}

// 删除专业
export function deleteMajor(majorId) {
  return request({
    url: `/graduate/majors/${majorId}`,
    method: 'delete'
  })
}

// ============ 分数线管理接口 ============

// 添加分数线
export function createScoreLine(data) {
  return request({
    url: '/graduate/score-lines',
    method: 'post',
    data
  })
}

// 更新分数线
export function updateScoreLine(scoreLineId, data) {
  return request({
    url: `/graduate/score-lines/${scoreLineId}`,
    method: 'put',
    data
  })
}

// 删除分数线
export function deleteScoreLine(scoreLineId) {
  return request({
    url: `/graduate/score-lines/${scoreLineId}`,
    method: 'delete'
  })
}

// ============ 考试科目管理接口 ============

// 添加考试科目
export function createExamSubject(data) {
  return request({
    url: '/graduate/exam-subjects',
    method: 'post',
    data
  })
}

// 更新考试科目
export function updateExamSubject(subjectId, data) {
  return request({
    url: `/graduate/exam-subjects/${subjectId}`,
    method: 'put',
    data
  })
}

// 删除考试科目
export function deleteExamSubject(subjectId) {
  return request({
    url: `/graduate/exam-subjects/${subjectId}`,
    method: 'delete'
  })
}
